src/identity-provider.js

```diff
@@ -0,0 +1,37 @@
+
+const isFunction = require('./utils/is-function')
+
+class IdentityProvider {
```

Comment:
I'll defer commenting on this for later as we get a little further with the Identity and address the comments so far. Eventually, we should take it towards something that *provides identities* as opposed to being a wrapper for signing and verifying *entries (=data)*.